homeassistant/components/weatherflow_cloud/weather.py

```diff
@@ -0,0 +1,145 @@
+"""Support for WeatherFlow Forecast weather service."""
+from __future__ import annotations
+
+from weatherflow4py.models.unified import WeatherFlowData
+
+from homeassistant.components.weather import (
+    Forecast,
+    SingleCoordinatorWeatherEntity,
+    WeatherEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    UnitOfPrecipitationDepth,
+    UnitOfPressure,
+    UnitOfSpeed,
+    UnitOfTemperature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ATTR_ATTRIBUTION, DOMAIN, MANUFACTURER
+from .coordinator import WeatherFlowCloudDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add a weather entity from a config_entry."""
+    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+
+    async_add_entities(
+        [
+            WeatherFlowWeather(coordinator, station_id=station_id)
+            for station_id, data in coordinator.data.items()
+        ]
+    )
+
+
+class WeatherFlowWeather(
+    SingleCoordinatorWeatherEntity[WeatherFlowCloudDataUpdateCoordinator]
+):
+    """Implementation of a WeatherFlow weather condition."""
+
+    _attr_attribution = ATTR_ATTRIBUTION
+    _attr_has_entity_name = True
+
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_native_precipitation_unit = UnitOfPrecipitationDepth.MILLIMETERS
+    _attr_native_pressure_unit = UnitOfPressure.MBAR
+    _attr_native_wind_speed_unit = UnitOfSpeed.METERS_PER_SECOND
+    _attr_supported_features = (
+        WeatherEntityFeature.FORECAST_DAILY | WeatherEntityFeature.FORECAST_HOURLY
+    )
+
+    def __init__(
+        self,
+        coordinator: WeatherFlowCloudDataUpdateCoordinator,
+        station_id: int,
+    ) -> None:
+        """Initialise the platform with a data instance and station."""
+        super().__init__(coordinator)
+
+        self.station_id = station_id
+        self._attr_unique_id = f"weatherflow_forecast_{station_id}"
+        self._attr_name = self.local_data.station.name
```

Comment:
On naming: I'd suggest naming the device to the station name. Then you can set `_attr_name = None`.